### PR TITLE
Make input `username` and `password` optional

### PR DESCRIPTION
**Pull request by Divyansh42**
Signed-off-by: divyansh42 <diagrawa@redhat.com>

### Description
- User may authenticate to image registry before running this action,
so in that case, `username` and `password` input is no more required.
- Add cron in workflows

### Related issue
Close: #36 
### Checklist

- [x] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is a patch change
- [x] This change is a minor change
- [ ] This change is a major (breaking) change

### CODE REVIEW

**Review by tetchel — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -158,7 +158,10 @@ async function run(): Promise<void> {
 
     const registryWithoutTrailingSlash = registry.replace(/\/$/, "");
 
-    const creds = `${username}:${password}`;
+    let creds = "";
+    if (username && password) {
```
Comment (Divyansh42) [inline src/index.ts:162]: I am wondering if we need to handle the case here when user has not provided only `username` but not `password` or vice versa.
I this case we can log a message (may be a warning message) to say something like:
`username is provided but password is missing`

File: .github/workflows/verify-push.yaml [hunk 2]
```diff
@@ -84,6 +86,16 @@ jobs:
             spring-petclinic-*.jar
           port: 8080
           oci: 'true'
+        env:
```
Comment (Divyansh42) [inline .github/workflows/verify-push.yaml:89]: will remove this once we merge the fix for buildah issue.

File: src/index.ts (hunk 1 above)
Comment (tetchel) [reply src/index.ts:162]: Yes, that's worth adding.

File: README.md [hunk 3]
```diff
@@ -124,15 +70,20 @@ jobs:
         dockerfiles: |
           ./Dockerfile
 
-    - name: Push To Quay
+    - name: Log in to quay.io
```
Comment (tetchel) [inline README.md:73]: Let's leave podman-login out of the example, but put a note in a comment that points to it.

File: .github/workflows/verify-push.yaml [hunk 4]
```diff
@@ -84,6 +86,16 @@ jobs:
...
+          STORAGE_OPTS: "overlay.mount_program=/usr/bin/fuse-overlayfs"
+
+      # Authenticate to container image registry to push the image
+      - name: Podman Login
```
Comment (tetchel) [inline .github/workflows/verify-push.yaml:93]: We have to have workflow tests that use username & password too, so we should have two workflows (or jobs), one with login and one with PTR username/password.

Comment (Divyansh42) [reply .github/workflows/verify-push.yaml:93]: Yeah, I have left one workflow with username/password approach. This one https://github.com/redhat-actions/push-to-registry/blob/main/.github/workflows/multiple-build.yaml
May be this is hard to figure out for users, shall I add new example?

Comment (tetchel) [reply .github/workflows/verify-push.yaml:93]: yeah, maybe an `login-example` and just `example`

File: README.md [hunk 5]
```diff
@@ -9,80 +9,24 @@
...
-      Separate arguments by newline. Do not use quotes.</td>
-  </tr>
-
-</table>
+| Input Name | Description | Default |
+| ---------- | ----------- | ------- |
+| image	| Name of the image you want to push. | **Required**
+| tags | The tag or tags of the image to push. For multiple tags, seperate by a space. For example, `latest ${{ github.sha }}` | `latest`
+| registry | URL of the registry to push the image to. Eg. `quay.io/<username>` | **Required**
+| username | Username with which to authenticate to the registry. | None
```
Comment (tetchel) [inline README.md:25]: we can add a note to username and password that says "required unless already logged in to the registry"

File: README.md [hunk 6]
```diff
@@ -97,7 +41,7 @@ For example, `[ quay.io/username/spring-image:v1, quay.io/username/spring-image:
 
 ## Examples
 
-The example below shows how the `push-to-registry` action can be used to push an image created by the [buildah-build](https://github.com/redhat-actions/buildah-build) action.
+The example below shows how the `push-to-registry` action can be used to push an image created by the **[buildah-build](https://github.com/redhat-actions/buildah-build)** action. **[podman-login](https://github.com/redhat-actions/podman-login)** action is used here to authenticate to quay.io to push the built image.
```
Comment (tetchel) [inline README.md:44]: Mention podman-login in the very top section too

File: README.md [hunk 7]
```diff
@@ -124,15 +70,20 @@ jobs:
         dockerfiles: |
           ./Dockerfile
 
-    - name: Push To Quay
+    - name: Log in to quay.io
+      uses: redhat-actions/podman-login@v1
```
Comment (tetchel) [inline README.md:74]: Remove this

File: README.md [hunk 8]
```diff
@@ -124,15 +73,18 @@ jobs:
         dockerfiles: |
           ./Dockerfile
 
-    - name: Push To Quay
+    # Podman Login action (https://github.com/redhat-actions/podman-login)ccan be used
```
Comment (tetchel) [inline README.md:76]: ccan -> can

File: src/index.ts [hunk 9]
```diff
@@ -158,7 +158,16 @@ async function run(): Promise<void> {
 
     const registryWithoutTrailingSlash = registry.replace(/\/$/, "");
 
-    const creds = `${username}:${password}`;
+    let creds = "";
+    if (username && !password) {
+        core.warning("Username is provided, but password is missing!");
```
Comment (tetchel) [inline src/index.ts:163]: Do not end the warnings with "!"

File: README.md [hunk 10]
```diff
@@ -1,88 +1,35 @@
...
 [![Link checker](https://github.com/redhat-actions/push-to-registry/workflows/Link%20checker/badge.svg)](https://github.com/redhat-actions/push-to-registry/actions?query=workflow%3A%22Link+checker%22)
 <br><br>
 [![tag badge](https://img.shields.io/github/v/tag/redhat-actions/push-to-registry)](https://github.com/redhat-actions/push-to-registry/tags)
 [![license badge](https://img.shields.io/github/license/redhat-actions/push-to-registry)](./LICENSE)
 [![size badge](https://img.shields.io/github/size/redhat-actions/push-to-registry/dist/index.js)](./dist)
 
-Push-to-registry is a GitHub Action for pushing a container image to an image registry, such as Dockerhub, Quay&#46;io, the GitHub Container Registry, or an OpenShift integrated registry.
+Push-to-registry is a GitHub Action for pushing a container image to an image registry, such as Dockerhub, quay&#46;io, the GitHub Container Registry, or an OpenShift integrated registry.
```
Comment (tetchel) [inline README.md:13]: why is the . escaped? :thinking:

**Review by tetchel — APPROVED** [review]
some very minor things and then we are good.

File: README.md (hunk 10 above)
Comment (Divyansh42) [reply README.md:13]: if . is not escaped it will come like a hyperlink.

**Review by tetchel — APPROVED** [review]

**Pull request merged**